server/src/infra/repositories/asset.repository.ts

```diff
@@ -830,9 +831,13 @@ export class AssetRepository implements IAssetRepository {
       .innerJoin('exif', 'e', 'asset."id" = e."assetId"')
       .leftJoin('smart_info', 'si', 'si."assetId" = asset."id"')
       .andWhere(
-        `(e."exifTextSearchableColumn" || COALESCE(si."smartInfoTextSearchableColumn", to_tsvector('english', '')))
-        @@ PLAINTO_TSQUERY('english', :query)`,
-        { query },
+        new Brackets((qb) => {
+          qb.where(
+            `(e."exifTextSearchableColumn" || COALESCE(si."smartInfoTextSearchableColumn", to_tsvector('english', '')))
+          @@ PLAINTO_TSQUERY('english', :query)`,
+            { query },
+          ).orWhere('asset."originalFileName" = :path', { path: path.parse(query).name });
```

Comment:
I'm not sure if this is where "path parsing" should be happening. It seems like something that should happen in the domain and be passed as a specific input into this method instead.